packages/nvidia_nat_test/src/nat/test/llm.py

```diff
@@ -0,0 +1,207 @@
+    )
+    delay_ms: int = Field(default=0, ge=0, description="Artificial per-call delay in milliseconds to mimic latency")
+
+
+class _ResponseChooser:
+    """
+    Helper class to choose the next response according to config using itertools.cycle and provide synchronous
+    and asynchronous sleep functions.
+    """
+
+    def __init__(self, response_seq: list[str], delay_ms: int):
+        self._cycler = iter_cycle(response_seq) if response_seq else None
+        self._delay_ms = delay_ms
+
+    def next_response(self) -> str:
+        """Return the next response in the cycle, or an empty string if no responses are configured."""
+        if self._cycler is None:
+            return ""
+        return next(self._cycler)
+
+    def sync_sleep(self) -> None:
+        time.sleep(self._delay_ms / 1000.0)
+
+    async def async_sleep(self) -> None:
+        await asyncio.sleep(self._delay_ms / 1000.0)
+
+
+@register_llm_provider(config_type=TestLLMConfig)
+async def test_llm_provider(config: TestLLMConfig, builder: Builder):
+    del builder  # suppress linting error
+    yield LLMProviderInfo(config=config, description="Test LLM provider")
+
+
+@register_llm_client(config_type=TestLLMConfig, wrapper_type=LLMFrameworkEnum.LANGCHAIN)
+async def test_llm_langchain(config: TestLLMConfig, builder: Builder):
+    """LLM client for LangChain"""
+    from langchain_core.messages import AIMessage
+
+    del builder  # suppress linting error
+    chooser = _ResponseChooser(response_seq=config.response_seq, delay_ms=config.delay_ms)
+
+    class LangChainTestLLM:
+
+        def invoke(self, messages: Any, **_kwargs: Any) -> AIMessage:
+            chooser.sync_sleep()
+            return AIMessage(content=chooser.next_response())
+
+        async def ainvoke(self, messages: Any, **_kwargs: Any) -> AIMessage:
+            await chooser.async_sleep()
+            return AIMessage(content=chooser.next_response())
+
+        def stream(self, messages: Any, **_kwargs: Any) -> Iterator[AIMessage]:
+            chooser.sync_sleep()
+            yield AIMessage(content=chooser.next_response())
+
+        async def astream(self, messages: Any, **_kwargs: Any) -> AsyncGenerator[AIMessage, None]:
+            await chooser.async_sleep()
+            yield AIMessage(content=chooser.next_response())
+
+    yield LangChainTestLLM()
+
+
+@register_llm_client(config_type=TestLLMConfig, wrapper_type=LLMFrameworkEnum.LLAMA_INDEX)
+async def test_llm_llama_index(config: TestLLMConfig, builder: Builder):
+    """LLM client for LlamaIndex"""
+    del builder  # suppress linting error
+    chooser = _ResponseChooser(response_seq=config.response_seq, delay_ms=config.delay_ms)
+
+    class LIChatMessage:
+
+        def __init__(self, content: str) -> None:
+            self.content = content
+
+    class LIChatResponse:
+
+        def __init__(self, text: str) -> None:
+            self.message = LIChatMessage(text)
+            self.text = text
```

Comment:
It is valid to mock a response like this without using the "proper" Llama Index types?